plugins/modules/postgresql_pg_hba.py

```diff
@@ -1307,50 +1367,80 @@ def main():
                     else:
                         # use module defaults
                         rule[key] = argument_spec[key].get('default', None)
-            new_rules.append(rule)
-        rules = new_rules
 
-    for rule in rules:
-        if rule.get('contype', None) is None:
-            continue
+            new_rules.append(rule)
+    # TODO
```

Comment:
```suggestion
    # TODO
```
does it make sense to move these lines to a separate function for readability?

Reply:
yes, it does, but I would do this in a later PR